iron-request.html

```diff
@@ -294,7 +294,10 @@
       }
 
       xhr.withCredentials = !!options.withCredentials;
-      xhr.timeout = options.timeout;
+
+      if (options.async) {
```

Comment:
You could move this line inside the if statement few lines above, inside the  `if (options.async !== false)`